### DISCUSSION TITLE
[Feature] hierarchical tags for navigation

### DISCUSSION THREAD

**Discussion by avandeputte** [immich-app/immich#2168] [Feature Request]
### The feature

I would like to be to be able to browse through photos by navigating through a tag hierarchy.

Tools like digikam are great for managing and assigning these hierarchical tags.  For example, this is a snapshot of a portion of my own tag hierarchy:  

<img width="373" alt="Screenshot 2023-04-04 at 2 20 10 PM" src="https://user-images.githubusercontent.com/20116257/229882972-aece95e7-5733-4197-a87a-3638c928147d.png">

For example:  `Activity --> Basketball --> Freshman Basketball`
I also do things like:  `People --> LastName --> LastName FirstName` to organize people in hierarchies
These tag values are stored in the exif data as a single value with delimiters.   (for example `Activity/Basketball/Freshman Basketball`)

I understand the delimiter can vary based on the application that is used (/ or | are common),  or even configurable.  So the tag parsing should be similarly configurable (specify the delimiter used on hierarchical tag parsing)

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by enboig** [comment]
I have a huge digikam collection, taking it into account when uploading using CLI would be great.

**Comment by charles-997** [comment]
Agree with this feature idea.

I think before the UI of "filtering the timeline" can be achieved a few core functions are required first:

1. Immich to read EXIF and XMP keywords/tags from imported images.
2. Immich to have webUI ability to add new keywords/tags to photos and edit existing imported keywords/tags.


I have been using the read only import feature. The vast majority of my photo collection has either Lightroom keywords or Digikam tags saved to the metadata either directly on the image as EXIF or using XMP sidecars.  I can't figure out how to view or use any of this metadata within Immich

**Reply by jrasm91** [reply]
Immich can and does support reading sidecar files, although we don't currently have any code to save the data from for tags (yet). Before we can support editing tags, we'd need to implement writing data to the sidecar file, otherwise they'd get overridden during the next exif extraction.

**Reply by avandeputte** [reply]
I too have been using the read-only import,  and I've been encouraged by the progress being made there!

As soon as tags are read and displayed,  this will be a viable application for my use case.  Right now even face tags don't seem to be parsed,  nevermind my tag hierarchy.  Also,  hoping at some point we'll be able to navigate through the tag hierarchy (maybe this could be some sort of "virtual" album structure?)

**Reply by charles-997** [reply]
> maybe this could be some sort of "virtual" album structure?)

Totally agree.  I much prefer tags + "smart albums" over tags.

I define a smart album as an album that automatically includes any photo with the specified tags.  In Lightroom, this is called "Collections"

The user simply adds the relevant tags to the photo and then the smart albums will update and the photo will be shown in the album.

In my view, the most essential benefit of this method is that it mitigates the risks of software lock in.  The album structure now resides in the metadata of each photo and could (in theory) be read by various software platforms.

The features described above are the sole reason I am an Immich GitHub sponsor. I am hoping that Immich will one day support that feature set.

**Comment by kvi77** [comment]
It is this function that I lack for complete happiness.

**Comment by charles-997** [comment]
Any interest in crowdfunding this feature?

I started looking into Synology Photos as an alternative to bridge the gap while Immich development continues.  I realized the Synology path costs quite a bit and it is probably best to stick with FOSS.

Happy to donate $50 to whomever has the skillset and time to put together a PR for this.  Hoping a few others on this thread would be able to match my contribution :)

**Comment by aisbergde** [comment]
This feature would be great. I manage photo metadata with digiKam.

**Comment by cr0n76** [comment]
I think hierachical tags could fulfill the functionality of some other feature requests as well.

E.g. you could add hierachical tags resembling the folder structure the photo is in. That would fulfill https://github.com/immich-app/immich/discussions/4279 with respective tag-views / tag-albums.

additionally if you add e.g. the day of recording and/or the folder structure as above, it would fulfill https://github.com/immich-app/immich/discussions/4694 since the users can choose which "surrounding assets" they want to see by using the respective tag.

Edit: Additionally i think this would keep the UI clean, as there is no need for another "folder view button" or "see surroundings button". Only tags and functionality to see photos filtered by tags (and somehow enable to create albums / views regarding tags) need to be added.

**Reply by GD-Dal** [reply]
The ability to read EXIF and IPTC data directly from file is essential if you ask me.
But this has no meaning of course if they not are displayed in the information pane.
Even more perfect if the metadata is clickable to display all other pictures with the same metadata. 
And searchable
Piwigo has had these features for years.
It's the only thing missing keeping me from using Immich TBH

Consider this a feature request :)

**Comment by paaspaas00** [comment]
@alextran1502 is this in your queue guys? That would be pretty awesome!

**Reply by bo0tzz** [reply]
Other than the open PRs, we don't have a queue.

**Comment by bo0tzz** [comment]
Will be closed by #11980.
